iOS-sample-app/Showcases/Screen Views/Modals/DismissalDelegation.swift

```diff
@@ -0,0 +1,21 @@
+
+import UIKit
+
+protocol DismissalDelegate: UIViewController {
```

Comment:
As discussed together, I'd rather keep the `Delegate` terminology instead of `Observer` as it's a widely known one among iOS devs.

I originally wanted this delegation pattern to be more "complete" but it wasn't necessary, so I simplified it.